src/elements/defaultLayout.tsx

```diff
@@ -1,15 +1,30 @@
+import { Dispatch, SetStateAction } from "react";
+
+import { AuthData } from "@/types";
+
 import Footer from "@/components/footer";
 import Header from "@/components/header/header";
 
-type DefaultLayoutProps = {
+interface DefaultLayoutProps {
   children: React.ReactNode;
-};
+  authData: AuthData;
+  setAuthData: Dispatch<SetStateAction<AuthData>>;
+  isSignInOpen: boolean;
+  setIsSignInOpen: Dispatch<SetStateAction<boolean>>;
+  targetRoute: string;
+}
 
 // Define a layout component for routes with Header and Footer
-export default function DefaultLayout({ children }: DefaultLayoutProps) {
+export default function DefaultLayout({ children, authData, setAuthData, isSignInOpen, setIsSignInOpen, targetRoute }: DefaultLayoutProps) {
   return (
     <>
-      <Header />
+      <Header
+        authData={authData}
+        setAuthData={setAuthData}
```

Comment:
header must contain only single onAuthUser callback. All modal-state manipulations must be inside header